modules/improvedigitalBidAdapter.js

```diff
@@ -192,539 +265,454 @@ export const spec = {
-        errors = errors || [];
-        errors.push({
-          errorCode: impressionObject.errorCode,
-          adUnitId: impressionObject.adUnitId
-        });
-      } else {
-        if (returnIdMappings) {
-          returnObject.idMappings.push({
-            adUnitId: impressionObject.adUnitId,
-            id: impressionObject.impressionObject.id
-          });
-        }
-        bidRequestObject.bid_request.imp = bidRequestObject.bid_request.imp || [];
-        bidRequestObject.bid_request.imp.push(impressionObject.impressionObject);
-
-        let writeLongRequest = false;
-        const outputUri = baseUrl + encodeURIComponent(JSON.stringify(bidRequestObject));
-        if (outputUri.length > this.CONSTANTS.MAX_URL_LENGTH) {
-          writeLongRequest = true;
-          if (bidRequestObject.bid_request.imp.length > 1) {
-            // Pop the current request and process it again in the next iteration
-            bidRequestObject.bid_request.imp.pop();
-            if (returnIdMappings) {
-              returnObject.idMappings.pop();
-            }
-            counter--;
-          }
-        }
+    // GPID
+    normalizedBidRequest.gpid = deepAccess(bid, 'ortb2Imp.ext.gpid') ||
+      deepAccess(bid, 'ortb2Imp.ext.data.pbadslot') ||
+      deepAccess(bid, 'ortb2Imp.ext.data.adserver.adslot');
 
-        if (writeLongRequest ||
-            !requestParameters.singleRequestMode ||
-            counter === impressionObjects.length - 1) {
-          returnObject.requests.push(this.formatRequest(requestParameters, bidRequestObject));
-          bidRequestObject = {
-            bid_request: this.createBasicBidRequestObject(requestParameters, extraRequestParameters)
-          };
-        }
-      }
-    }
+    // Interstitial Signal
+    normalizedBidRequest.interstitialSignal = deepAccess(bid, 'ortb2Imp.instl');
 
-    if (errors) {
-      returnObject.errors = errors;
-    }
+    return normalizedBidRequest;
+  },
 
-    return returnObject;
-  };
+  isValidSize(sizePair) {
+    return sizePair.length === 2 &&
+      isInteger(sizePair[0]) &&
+      isInteger(sizePair[1]) &&
+      sizePair[0] >= 0 &&
+      sizePair[1] >= 0;
+  }
+};
 
-  this.formatRequest = function(requestParameters, bidRequestObject) {
-    switch (requestParameters.returnObjType) {
-      case this.CONSTANTS.RETURN_OBJ_TYPE.URL_PARAMS_SPLIT:
-        return {
-          method: 'GET',
-          url: `${requestParameters.adServerBaseUrl}/${this.CONSTANTS.END_POINT}`,
-          data: `${this.CONSTANTS.AD_SERVER_URL_PARAM}${encodeURIComponent(JSON.stringify(bidRequestObject))}`
-        };
-      default:
-        const baseUrl = `${requestParameters.adServerBaseUrl}/` +
-          `${this.CONSTANTS.END_POINT}?${this.CONSTANTS.AD_SERVER_URL_PARAM}`;
-        return {
-          url: baseUrl + encodeURIComponent(JSON.stringify(bidRequestObject))
-        }
-    }
-  };
+const ID_REQUEST = {
+  buildImpression(bidRequest) {
```

Comment:
please rename to `buildImp`. Many people get confused by impression meaning the ad impression when showing the ad.